acquisition/tracking.py

```diff
@@ -2,208 +2,384 @@
+        self.VIEW_TYPE = "image"        # "image" to block out white binary noise, "binary" to block out black binary noise
+        self.START_FRAME = 0            # Defines starting frame. ONLY FOR DEBUGGING
+        self.FPS = 20                   # FPS of the camera
+        self.START_VOLTAGE = 40         # Initial voltage value 
+        self.VOLTAGE_INCREMENT = 5      # Voltage step between datapoints
+        self.CHANGE_INTERVAL = 5        # Time between data points in the real-time trial (seconds)
+        self.SAMPLE_FRAMES = 15         # Number of frames averaged over per data point
+        self.BIN_THRESH = 26            # Binary threshold for object detection
+        self.X_RANGE = (200, 900)      # x-axis frame of interest limits
+        self.Y_RANGE = (554, 1000)      # y-axis frame of interest limits
+        self.BOTTOM_BAR = 100           # Erasure rectangles, measured in pixels from the corresponding edge
+        self.TOP_BAR = 0
+        self.LEFT_BAR = 0
+        self.RIGHT_BAR = 0
+        self.PIXELCONVERSION = 0.01628  # Pixel-to-millimeter conversion, gathered from calibration image. Set to 1 to output the rawpixel data
 
-# --------------------------- Constants Definition ---------------------------------------------- #
-VIDEO_FILE = "../ExampleSplit.avi"
-VIEW_TYPE = "image"  # Options: "binary", "image", "frame", "cleanthresh"
-OUTPUT = "none"      # Options: "tuple", "none"
-SHOW_QUANT = "none"  # Options: "height", "micromotion", "both"
-POOR_TRACKING = False # Ignoring index of the particle if varying
-FPS = 20
-CHANGE_INTERVAL = 5
-POINTS_TO_IMAGE = [] # Frames to capture
-SAMPLE_FRAMES = 15
-BIN_THRESH = 26
-X_RANGE = (800, 1200)
-Y_RANGE = (547, 933)
-BOTTOM_BAR = 100
-TOP_BAR = 0
-LEFT_BAR = 0
-RIGHT_BAR = 0
-INDEX_LIST = [] # Indices of desired tracking particles
-
-# ------------------------------- Helper Functions ---------------------------------------------- #
-
-def initialize_video(cap):
-    total_frames = int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
-    cleaning_kernel = np.ones((2, 2), np.uint8)
-    filling_kernel = np.ones((2, 2), np.uint8)
-    return total_frames, cleaning_kernel, filling_kernel
-
-def setup_detector():
-    params = cv2.SimpleBlobDetector.Params()
-    # Filter by color
-    params.filterByColor = True
-    params.blobColor = 255
-    # Filter by area (pixels)
-    params.filterByArea = True
-    params.minArea = 5
-    params.maxArea = 300
-    # Filter by circularity
-    params.filterByCircularity = False
-    # Filter by convexity
-    params.filterByConvexity = False
-    # Filter by inertia ratio (To detect elongated shapes)
-    params.filterByInertia = False
-    params.minInertiaRatio = 0.01
-    params.maxInertiaRatio = 0.3
-    return cv2.SimpleBlobDetector_create(params)
-
-def get_frame(cap, got_frame_num):
-    cap.set(cv2.CAP_PROP_POS_FRAMES, got_frame_num)
-    ret, got_frame = cap.read()
-    return ret, got_frame
 
```

Comment:
Instead of declaring config as a global variable, create a function

def get_default_config():
    return TrackingConfig()

and then every method that uses config should just get a keywork argument like this:

def frame_dimensions(cap, frame_num, config=get_default_config()):


This way you don't need to change the code in every instance every method is used.